src/main/java/io/github/thebusybiscuit/slimefun4/core/services/CustomItemDataService.java

```diff
@@ -45,7 +45,14 @@ public void setItemData(ItemMeta im, String id) {
     }
 
     public Optional<String> getItemData(ItemStack item) {
-        return getItemData(item.getItemMeta());
+        if (!item.hasItemMeta()) {
+            return Optional.empty();
+        }
+        ItemMeta meta = item.getItemMeta();
+        if (meta == null) {
```

Comment:
This isn't ever null

Reply:
it could be for wrappers

Reply:
No in the wrapper you throw an exception now

Reply:
In that case this code won't even be reached though since we prevent this from ever being called due to the #hasItemMeta() check from before.

Reply:
busybiscuit told me to explicitly add a check here, that is the only reason I put it

Reply:
You only seem to read half of our comments then, I already told you this is redundant and that it isn't nullable. Only the actual ItemStack is.

Reply:
I am reading all your comments, you are just confusing, edits are enabled for maintainers, please change the PR to be as you see fit

Reply:
Tell us what's unclear then. There is nothing that can't be discussed or cleared up.